pwiz_tools/Skyline/Model/SrmDocument.cs

```diff
@@ -2115,10 +2115,15 @@ public double GetCollisionEnergy(PeptideDocNode nodePep, TransitionGroupDocNode
         public static double GetCollisionEnergy(SrmSettings settings, PeptideDocNode nodePep,
             TransitionGroupDocNode nodeGroup, TransitionDocNode nodeTran, CollisionEnergyRegression regression, int step)
         {
-            var ce = nodeTran==null // If we're only given a precursor, use the explicit CE of its children if they all agree
-                ? (nodeGroup.Children.Any() && nodeGroup.Children.All( node => ((TransitionDocNode)node).ExplicitValues.CollisionEnergy == ((TransitionDocNode)nodeGroup.Children.First()).ExplicitValues.CollisionEnergy) 
-                    ? ((TransitionDocNode)nodeGroup.Children.First()).ExplicitValues.CollisionEnergy : null)
-                : nodeTran.ExplicitValues.CollisionEnergy;
+            // Collision Energy explicitly declared at the transition level is taken to be the correct value.

```

Comment:
This is a mess. Can it be broken out as a separate function using ifs and return statements, which would also make it more debuggable.

Reply:
Its in a pretty brief function already, so I've just changed it in place use if statements.